### DISCUSSION TITLE
Handling large commits

### DISCUSSION THREAD

**Discussion by gravitydeepimpactsun** [Blarc/ai-commits-intellij-plugin#113] [Ideas]
Rather than submitting the entire diff to GPT, what about if each changed file's diff was sent individually then send all of the generated commit messages to GPT together with a prompt to condense into a single message highlighting what GPT thinks is most relevant?

(JetBrains AI Assistant seems to do this)

**Comment by Blarc** [comment]
There was already some debate in https://github.com/Blarc/ai-commits-intellij-plugin/issues/10 regarding this.
The biggest problem I see with this way of handling large commits is that this would be quite expensive. It would definitely need to be optional.
